libmamba/include/mamba/util/url.hpp

```diff
@@ -57,99 +61,156 @@ namespace mamba::util
         [[nodiscard]] auto scheme() const -> const std::string&;
 
         /** Set a non-empty scheme. */
-        auto set_scheme(std::string_view scheme) -> URL&;
+        void set_scheme(std::string_view scheme);
```

Comment:
It's annoying to lose the chained construction of the URL. I wonder if it could make sense to use a CRTP here, or even to merge the CondaURL and the URL classes? (do we have usages of URL where we don't want to have access to the additiona methods form CondaURL?

Reply:
After discussion in person:
- having two distinct abstractions for URL and CondaURL is better thatn a single class; it helps to keep isolation of the dependency on CURL
- CRTP is a bit of overkill just for keeping the chained construction syntax.